doc/analyzers/VSTHRD201.md

```diff
@@ -0,0 +1,83 @@
+# VSTHRD201 Check cancellation after `SwitchToMainThreadAsync`
+
+Code that should not run if a CancellationToken is canceled should call
+`CancellationToken.ThrowIfCancellationRequested()` explicitly rather than
+relying on `JoinableTaskFactory.SwitchToMainThreadAsync(CancellationToken)` to
+throw `OperationCanceledException`.
+
+While passing a `CancellationToken` to `SwitchToMainThreadAsync` is an effective
+way to abort a switch to the main thread that is taking too long, it is not a guarantee
+that code execution will not continue when the token is canceled.
+
+The `SwitchToMainThreadAsync(CancellationToken)` method *will* throw `OperationCanceledException`
+back at the caller if **all** these conditions are met:
+
+1. The `CancellationToken` passed to it is canceled.
+2. The caller was not on the main thread already.
+3. The request to resume the main thread is fulfilled by the threadpool before the main thread can respond to the request.
```

Comment:
💭 It seems like these are slightly off. If (1) and (2) are met, the request will be processed synchronously and not depend on (3). If the cancellation occurs after the call is made to `SwitchToMainThreadAsync`, then (3) comes into play.

Reply:
I think you misread no 2. It will *not* complete synchronously when 2 is met.

Reply:
Filed #569 to track separately as a library bug. The docs as it is in this PR are correct.